### PR TITLE
Feature: Add more tests

### PR DESCRIPTION
**Pull request by nagavcindriqim**
Added some tests, increased PovioKit test coverage from 62.5% to 66.5%.

### CODE REVIEW

File: Tests/Tests/Core/Extensions/Foundation/OptionalTests.swift [hunk 1]
```diff
@@ -38,4 +38,14 @@ class OptionalTests: XCTestCase {
...
   }
+  
+  func test_replacingString_producesExpectedResult() {
+    let sut: String? = "Nice day today!"
+    XCTAssertEqual(sut.replacingString(with: "night", range: NSRange(location: 5, length: 3)), "Nice night today!")
+  }
+  
+  func test_replacingString_doesNothingOnNilValue() {
+    let sut: String? = nil
+    XCTAssertNotEqual(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)), "Nice night today!")
```
Comment (yllfejziu) [inline Tests/Tests/Core/Extensions/Foundation/OptionalTests.swift:49]: Would it make more sense to make this test asserting for nil value like this:
`XCTAssertNil(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)))`

File: Tests/Tests/Core/Extensions/UIKit/UIViewControllerTests.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+//  Created by Ndriqim Nagavci on 27/10/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+import PovioKit
+import PovioKitUI
+
+class UIViewControllerTests: XCTestCase {
+  func test_createAllButtonTypes() {
```
Comment (yllfejziu) [inline Tests/Tests/Core/Extensions/UIKit/UIViewControllerTests.swift:14]: Please update the function name to match the pattern with the functions below: `test_x_doesY`

File: Tests/Tests/Core/Utilities/AttributedStringBuilder/AttributedStringBuilderTests.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+@testable import PovioKit
+
+class AttributedStringBuilderTests: XCTestCase {
+  let text = "My custom text!"
+  let substring = "custom"
+  let textColor: UIColor = .red
+  let font: UIFont = .boldSystemFont(ofSize: 14)
+  let underlineStyle: NSUnderlineStyle = .thick
+  let paragraphLineSpacing: CGFloat = 10
+  let paragraphLineHeight: CGFloat = 30
```
Comment (yllfejziu) [inline Tests/Tests/Core/Utilities/AttributedStringBuilder/AttributedStringBuilderTests.swift:19]: Would it be better if we move all these helper properties below in an extension?
e.g.
`MARK: - Helpers
extension AttributedStringBuilderTests {
  func anyText() -> String {
    "My custom text!"
  }
}

File: Tests/Tests/Core/Extensions/Foundation/OptionalTests.swift (hunk 1 above)
Comment (nagavcindriqim) [reply Tests/Tests/Core/Extensions/Foundation/OptionalTests.swift:49]: We can use:
`XCTAssertNotNil(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)))`
Since `replacingString` returns non-optional String

File: Tests/Tests/Core/Utilities/AttributedStringBuilder/AttributedStringBuilderTests.swift (hunk 3 above)
Comment (nagavcindriqim) [reply Tests/Tests/Core/Utilities/AttributedStringBuilder/AttributedStringBuilderTests.swift:19]: Maybe we can use enum in this case?
`// MARK: - Helpers
private extension AttributedStringBuilderTests {
  enum Values {
    static let text = "My custom text!"
  }
}

Thoughts?

Comment (yllfejziu) [reply Tests/Tests/Core/Utilities/AttributedStringBuilder/AttributedStringBuilderTests.swift]: Looks good.

**Review by yllfejziu — APPROVED** [review]
👌
